util/monitoring.js

```diff
@@ -0,0 +1,111 @@
+const _ = require('lodash');
+const googleMonitoring = require('@google-cloud/monitoring');
+const fs = require('fs');
+
+function getValueFromPoint(point) {
+  const valueKey = _.get(point, 'value.value');
+  return _.get(point, `value.${valueKey}`);
+}
+
+async function requestAndAssembleInfo(client, request) {
+  const [timeSeriesList] = await client.listTimeSeries(request);
+  if (timeSeriesList.length === 0) {
+    return {
+      err: `Can't find information`,
+    };
+  }
+  const timeSeries = timeSeriesList[0];
+  const points = _.get(timeSeries, 'points');
+  const info = {};
+  info.start = getValueFromPoint(points[0]);
+  info.end = getValueFromPoint(points[points.length - 1]);
+  info.min = getValueFromPoint(_.minBy(points, (p) => {
+    return getValueFromPoint(p);
+  }));
+  info.max = getValueFromPoint(_.maxBy(points, (p) => {
+    return getValueFromPoint(p);
+  }));
+  return info;
+}
+
+async function getCpuUsageInfo(client, projectId, instanceName, startTime, endTime) {
+  const filter = `metric.labels.instance_name = "${instanceName}" AND metric.type="compute.googleapis.com/instance/cpu/utilization"`;
+  const request = {
+    name: client.projectPath(projectId),
+    filter: filter,
+    interval: {
```

Comment:
minor: Can this block be extracted as a separate function and reused?

Reply:
Looks better! Done